backend/shared/shared/common/tests/factories.py

```diff
@@ -9,7 +9,7 @@ class UserFactory(factory.django.DjangoModelFactory):
 
     first_name = factory.Faker("first_name")
     last_name = factory.Faker("last_name")
-    username = factory.Faker("user_name")
+    username = factory.Faker("uuid4")
```

Comment:
As the Helsinki-profile UUID is saved as `username` it makes sense that our mock-data uses UUID there as well. BTW. @karisal-anders you know why is it like that? :D

Reply:
Helsinki-profile has two different UUID values:
 - Helsinki profile ID
 - User ID

The shared backend's HelsinkiOIDCAuthenticationBackend.create_user will create a Django user whose username contains the [sub claim](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims) of the OIDC which is the user ID, not the Helsinki profile ID. The Helsinki profile ID is not in the OIDC claims at all but can be read using some other query—HELSINKI_PROFILE_API_URL is probably related to this but might not currently be used for that purpose but for getting a user's national identity number instead AFAIK [1]. Some discussion on the profile ID and user ID in city of Helsinki's slack channel #helsinki-profiili: [#1](https://helsinkicity.slack.com/archives/CB3HG29K8/p1669637782539189?thread_ts=1669635753.777089&cid=CB3HG29K8), [#2](https://helsinkicity.slack.com/archives/CB3HG29K8/p1669638242415039?thread_ts=1669635753.777089&cid=CB3HG29K8).

[1] backend/README.md:
> call `settings.HELSINKI_PROFILE_API_URL` with the access token [to get the national identification number](https://github.com/City-of-Helsinki/yjdh/blob/develop/backend/shared/shared/helsinki_profile/hp_client.py#L24)

Reply:
I don't know why the Helsinki profile's user ID—not the profile ID but the user ID—is passed in the sub claim, this documentation may possibly shed some light on this matter: https://github.com/City-of-Helsinki/open-city-profile/blob/develop/open_city_profile/static/open-city-profile/gdpr-api-openapi.yaml

Reply:
About changing the UserFactory's username to uuid4 for all use cases... hmm... I'm not sure whether all paths where users are created use username that is uuid4. Maybe it would be better to have e.g. `HelsinkiProfileUserFactory` which would be:
```python
class HelsinkiProfileUserFactory(UserFactory):
    username = factory.Faker("uuid4")  # Helsinki profile linked users' username is an uuid4
```
and keep the `UserFactory` as is. What do you think?